aggregator.c

```diff
@@ -237,18 +237,10 @@ aggregator_putmetric(
 			if (invocation->hash == omhash && \
 					strcmp(o, invocation->metric) == 0)  /* match */ \
 				break;
-		pthread_rwlock_rdlock(&compute->invlock);
+		pthread_rwlock_wrlock(&compute->invlock);
 		find_invocation(ometric);
 		/* switch to a write lock from here, since we've found what we
 		 * were looking for (or are going to create it) and modify it */
```

Comment:
if you start with a write-lock, then this comment should go as well, since we aint switching any more.